tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -532,20 +532,22 @@ def do_rolling_bounce(self, processor, counter, current_generation):
                         log_monitor.wait_until("Sent a version 8 subscription and got version 7 assignment back (successful version probing). Downgrade subscription metadata to commonly supported version 7 and trigger new rebalance.",
                                                timeout_sec=60,
                                                err_msg="Could not detect 'successful version probing' at upgrading node " + str(node.account))
-                    else:
-                        log_monitor.wait_until("Sent a version 8 subscription and got version 7 assignment back (successful version probing). Downgrade subscription metadata to commonly supported version 8 and trigger new rebalance.",
```

Comment:
This verification is only true if the two other processors haven't rebalanced before the processor that bounced last re-joins the group. If this rebalance happens or not depends also on how long the log files movements on lines 507-509 last.
Actually, the test `test_version_probing_upgrade` is independent of the used task assignor, but this issue was surfaced by the `HighAvailabilityTaskAssignor` but not by the `StickyTaskAssignor`. I cannot say for sure why.

Reply:
Thanks @cadonna ; I agree. This test should just be verifying that we first converge on 7, and then that we converge on 8.